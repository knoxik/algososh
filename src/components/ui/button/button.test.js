import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

import { Button } from './button';


const buttonActiveProps = {
    text: 'кнопка',
    isLoader: false
}

const buttonLoaderProps = {
    text: 'кнопка',
    isLoader: true
}

const buttonDisabledProps = {
    text: 'кнопка',
    isLoader: false,
    disabled: true
}

const buttonWithoutText = {
    isLoader: false,
}

const onClick = () => {
    alert('good')
}

describe('Проверка кнопки', () => {
    it('Кнопка активна', () => {
        const ButtonComponent = renderer.create(<Button {...buttonActiveProps}/>).toJSON();
        expect(ButtonComponent).toMatchSnapshot();
    }) 

    it('Кнопка не активна', () => {
        const ButtonComponent = renderer.create(<Button {...buttonDisabledProps}/>).toJSON();
        expect(ButtonComponent).toMatchSnapshot();
    }) 

    it('Лоадер на кнопке', () => {
        const ButtonComponent = renderer.create(<Button {...buttonLoaderProps}/>).toJSON();
        expect(ButtonComponent).toMatchSnapshot();
    }) 

    it('Кнопка без текста', () => {
        const ButtonComponent = renderer.create(<Button {...buttonWithoutText}/>).toJSON();
        expect(ButtonComponent).toMatchSnapshot();
    }) 

    it('Нажатие на кнопку вызывает корректный alert', () => {
        window.alert = jest.fn();
        render(<Button text='test' onClick={onClick}/>);

        const button = screen.getByText('test');
        fireEvent.click(button);
        expect(window.alert).toHaveBeenCalledWith('good');
    }) 
});