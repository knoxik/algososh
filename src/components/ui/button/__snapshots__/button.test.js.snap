// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка кнопки Кнопка активна 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    className="text undefined"
  >
    кнопка
  </p>
</button>
`;

exports[`Проверка кнопки Кнопка без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Проверка кнопки Кнопка не активна 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    кнопка
  </p>
</button>
`;

exports[`Проверка кнопки Лоадер на кнопке 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  data-testid="button"
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
